ev3/legoev3_motor.c

```diff
@@ -628,13 +633,22 @@ static void calculate_speed(struct legoev3_motor_data *ev3_tm)
 			ev3_tm->speed  = -new_speed;
 
 		ev3_tm->stalled = 0;
+		ev3_tm->stalling = 0;
 		ev3_tm->got_new_sample = false;
 
 	} else if (ev3_tm->max_us_per_sample < ktime_to_us(ktime_sub(ktime_get(), ev3_tm->tacho_samples[diff_idx]))) {
 
 		ev3_tm->dir_chg_samples = 0;
 		ev3_tm->speed = 0;
-		ev3_tm->stalled = 1;
+		ev3_tm->stalled = 0;
+
+		if (ev3_tm->stalling) {
+			if ((TACHO_MOTOR_STALLED_MS * USEC_PER_SEC) < ktime_to_us(ktime_sub(ktime_get(), ev3_tm->stalling_since)))
```

Comment:
Because I did not know it exists, and you use the same pattern a [couple of lines below](https://github.com/ddemidov/lego-linux-drivers/blob/3ef55b23c6a78d83b20fcdd642506e8612b6d5f9/ev3/legoev3_motor.c#L654) :)